src/XrdClient/XrdClient.cc

```diff
@@ -1372,15 +1372,15 @@ bool XrdClient::OpenFileWhenRedirected(char *newfhandle, bool &wasopen)
 	Info(XrdClientDebug::kHIDEBUG,
 	     "OpenFileWhenRedirected", "Stripping off the 'delete' option." );
 
-	options &= !kXR_delete;
```

Comment:
This does not seem right at the first look. `kXR_delete` is a positive integer (enum), which would be translated to `true` here and negating that would result in `false` (i.e. 0).